### ISSUE TITLE
NO_EPOLL=1 mode handles RDHUP incorrectly (in debug mode).

### ISSUE THREAD

**Issue by srh** [rethinkdb/rethinkdb#70]
Some code in poll.cc is inconsistent with the code in epoll.cc.  A debug-mode assertion in poll.cc should fail because it expects an event not to have rdhup as part of the bitmask.  Namely, the assertion at the top of `user_to_poll`.

Once the OS X port is done, this issue will be resolved, because we will no longer be using RDHUP.  Consider this low priority for that reason and for the reason that requires DEBUG=1 NO_EPOLL=1 to manifest.
